eth/src/main/kotlin/com/d3/eth/provider/EthRelayProviderIrohaImpl.kt

```diff
@@ -40,9 +41,13 @@ class EthRelayProviderIrohaImpl(
     }
 
     /** Get relay belonging to [irohaAccountId] */
-    override fun getRelay(irohaAccountId: String): Result<String, Exception> {
+    override fun getRelayByAccountId(irohaAccountId: String): Result<Optional<String>, Exception> {
         return getRelays().map { relays ->
-            relays.filter { it.value == irohaAccountId }.keys.first()
+            val filtered = relays.filter { it.value == irohaAccountId }.keys
+            if (filtered.isEmpty())
+                Optional.empty()
```

Comment:
Why we use `Optional` here? `String?` reduces the need for that.

Reply:
Good point, but unfortunately:
1. Kotlin Result cannot be used as return from function right now (https://stackoverflow.com/questions/52631827/why-cant-kotlin-result-be-used-as-a-return-type)
2. So, we use https://github.com/kittinunf/Result
3. The last result has type Result<Any, Exception> but not Result<Any?, Exception>, so nullable type cannot be used.
So, only option I see is to use Java Optional.